docs/architecture/adr-038-state-listening.md

```diff
@@ -3,6 +3,10 @@
 ## Changelog
 
 * 11/23/2020: Initial draft
+* 10/06/2022: Introduce plugin system based on hashicorp/go-plugin
```

Comment:
Are these all are addressed in this PR right? If so, then just have it under a single date/entry.